plugins/woocommerce-admin/client/products/sections/product-details-section.tsx

```diff
@@ -31,10 +36,13 @@ import { getCheckboxProps, getTextControlProps } from './utils';
 const PRODUCT_DETAILS_SLUG = 'product-details';
 
 export const ProductDetailsSection: React.FC = () => {
-	const { getInputProps, values, touched, errors } =
+	const { getInputProps, values, setValue, touched, errors } =
 		useFormContext< Product >();
 	const [ showProductLinkEditModal, setShowProductLinkEditModal ] =
 		useState( false );
+	const [ descriptionBlocks, setDescriptionBlocks ] = useState<
```

Comment:
With this we have to source of truth one inside this component and other inside the form context which can be tricky solving any issue related to final description value. Also if internal `description` value from inside the form context changed outside of this section component, the `descriptionBlocks` will never receive the update.

Reply:
>With this we have to source of truth one inside this component 

I understand what you mean, but that's not quite the case here.  The source of truth is from the props passed to the component.  Inside the component, we hold a `ref` that is updated whenever the `blocks` prop passed in changes.  The source of truth is always the prop passed in.  The ref is used for history tracking and optimization.

>Also if internal description value from inside the form context changed outside of this section component

That's a great point, though we don't really have a case for this at the moment, do we?  In order to do this, we'd probably want to keep another state variable for `description` inside this component and/or parse and memoize the blocks being passed in everytime.

Since we don't have a use case for this yet, I'm hesitant to complicate the logic further in this component for something we might not use.  But if you feel strongly about this or see a case where we set the description from another component inside the form, then I'll add this in.